sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -112,10 +112,12 @@ object QueryExecutionErrors extends QueryErrorsBase {
   }
 
   def invalidInputSyntaxForNumericError(
+      to: AbstractDataType,
```

Comment:
I don't know whether there is a better way than this `AbstractDataType` and corresponding `tosimpleString` method, considering the `DecimalType`. Since the cast fails, we can't get a concrete `DecimalType` class with scale and precision, so I just can't use the `DataType` here.

Reply:
Problem solved if we use the DecimalType from the caller of fromStringANSI:
https://github.com/apache/spark/pull/36244/files#diff-393a0ad195d66fb094f2dca3669c1317b7f57a6637f6e95241b57bc6550ce49aR819